index.coffee

```diff
@@ -0,0 +1,65 @@
+through = require 'through'
+request = require 'request'
+
+module.exports = ({ url, maxRetries = 10 }) ->
+	if not url?
+		throw new Error('resumable requires url parameter')
+	throughStream = through (data) ->
+		req.bytesRead += data.length
+		@emit('data', data)
+	req =
+		stream: throughStream
+		bytesRead: 0
+		retries: 0
+		maxRetries: maxRetries
+		contentLength: null
+		requestOpts:
+			url: url
+	init(req)
+	return req.stream
+
+# Send a HEAD request to get total Content-length
+# The "response" and "request" events are emitted on the resumable stream,
+# to mimic what request would do if there was only one (non-resumable) request.
+init = (req) ->
+	request.head req.requestOpts.url , (err, response) ->
```

Comment:
Could we get the total length by inspecting the first normal GET request instead of an additional HEAD request in the beginning?